update/src/Pos/Update/Poll/Failure.hs

```diff
@@ -22,87 +22,121 @@ import           Pos.Crypto (shortHashF)
-    | PollNotRichman { pnrStakeholder :: !StakeholderId
-                    ,  pnrThreshold   :: !Coin
-                    ,  pnrStake       :: !(Maybe Coin)}
-    | PollUnknownProposal { pupStakeholder :: !StakeholderId
-                         ,  pupProposal    :: !UpId}
+    -- | PollSmallProposalStake
+    --       pspsThreshold
+    --       pspsActual
+    --       pspsUpId
+    | PollSmallProposalStake !Coin !Coin !UpId
+    -- | PollNotRichman
+    --       pnrStakeholder
+    --       pnrThreshold
+    --       pnrStake
+    | PollNotRichman !StakeholderId !Coin !(Maybe Coin)
+    -- | PollUnknownProposal
+    --       pupStakeholder
+    --       pupProposal
+    | PollUnknownProposal !StakeholderId !UpId
     | PollUnknownStakes !EpochIndex
-    | PollWrongSoftwareVersion { pwsvStored :: !(Maybe NumSoftwareVersion)
-                              ,  pwsvApp    :: !ApplicationName
-                              ,  pwsvGiven  :: !NumSoftwareVersion
-                              ,  pwsvUpId   :: !UpId}
-    | PollProposalIsDecided { ppidUpId        :: !UpId
-                           ,  ppidStakeholder :: !StakeholderId}
-    | PollExtraRevote { perUpId        :: !UpId
-                     ,  perStakeholder :: !StakeholderId
-                     ,  perDecision    :: !Bool}
-    | PollWrongHeaderBlockVersion { pwhpvGiven   :: !BlockVersion
-                                  , pwhpvAdopted :: !BlockVersion}
-    | PollBadBlockVersion { pbpvUpId       :: !UpId
-                            ,  pbpvGiven   :: !BlockVersion
-                            ,  pbpvAdopted :: !BlockVersion}
-    | PollTooLargeProposal { ptlpUpId  :: !UpId
-                           , ptlpSize  :: !Byte
-                           , ptlpLimit :: !Byte
-                           }
-    | PollMoreThanOneProposalPerEpoch { ptopFrom :: !StakeholderId
-                                      , ptopUpId :: !UpId
-                                      }
-    | PollUnknownAttributesInProposal { puapUpId  :: !UpId
-                                      , puapAttrs :: !UpAttributes
-                                      }
-    | PollTipMismatch { ptmTipDB     :: !HeaderHash
-                      , ptmTipMemory :: !HeaderHash
-                      }
+    -- PollWrongSoftwareVersion
+    --    pwsvStored
+    --    pwsvApp
+    --    pwsvGiven
+    --    pwsvUpId
+    | PollWrongSoftwareVersion
+          !(Maybe NumSoftwareVersion)
+          !ApplicationName
+          !NumSoftwareVersion
+          !UpId
+    -- | PollProposalIsDecided
+    --      ppidUpId
+    --      ppidStakeholder
+    | PollProposalIsDecided !UpId !StakeholderId
+    -- | PollExtraRevote
+    --       perUpId
+    --       perStakeholder
+    --       perDecision
+    | PollExtraRevote !UpId !StakeholderId !Bool
+    -- | PollWrongHeaderBlockVersion
+    --       pwhpvGiven
+    --       pwhpvAdopted
+    | PollWrongHeaderBlockVersion !BlockVersion !BlockVersion
```

Comment:
Should we maybe use some `newtype`s for some of these constructor fields?

Reply:
or am I going overboard?

Reply:
This is an error constructor. The absolute worst thing that can happen here is that the error message is wrong. In this casned, I'm ok with leaving it as it is :). If this was something like a transaction with source and destination addresses, then a `newtype` would make a **lot** of sense.

Reply:
Agree with @erikd that it doesn’t seem critical in this case.

Reply:
@erikd @mhuesch Thanks guys